src/internal/analytics-metadata/dom-utils.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { METADATA_DATA_ATTRIBUTE } from './attributes';
+
+export const findNextNode = (node: HTMLElement): HTMLElement | null => {
+  try {
+    const referrer = node.dataset.awsuiReferrerId;
```

Comment:
Out of scope of this PR but if we have code in this package that refer to this attribute and code in the components package then maybe we need to have additional utilities in component-toolkit that define a constant value for this so that these cannot go out of sync (maybe very unlikely?)